pilot/pkg/security/authz/model/permission.go

```diff
@@ -212,7 +212,6 @@ func isSupportedPermission(key string) bool {
 	case key == attrDestIP:
 	case key == attrDestPort:
 	case key == pathHeader || key == methodHeader || key == hostHeader:
-	case strings.HasPrefix(key, attrRequestHeader):
```

Comment:
Does this change relate to this PR description?

Reply:
Not really related to this PR but this line is useless because the attrRequestHeader is already handled as part of the principal in isSupportedPrincipal().

Reply:
It's not clear from this change, but I trust your claims ;)

Reply:
Yeah, I double checked this.